### DISCUSSION TITLE
Idea for additional project - "Binary Image-team" (where else to post ideas?)

### DISCUSSION THREAD

**Discussion by zt3ph4n** [zephyrproject-rtos/zephyr#59276] [Q&A]
# Introduction of "Zephyr-RTOS Binary release-team"

Hi all! 

I've spotted a demand for a team/group with builders & testers, one or two coders for random implement-issues etc, focused to build images with varying settings/hardening/security-add's for various system, and should do this with three top priorities;
* Security.
* Compatibility.
* Broadreaching (build for diff laptops,routers,IoT generally,then maybe extending to phones & models).

What we can accomplish for common people is simply building the Zephyr-version's perhaps going to start some people on 2.x-thread,some for newer 3.3 until we can maintain updates more dynamically by custom tools and can setup a safe dump-site to let people send prebuilt system-images for inspection and crc-checking for missing system's before getting them dividided in the example to come, and also maintain updates as regurarly as possible - to strive towards 3-5 versions of each device built for in sub-groups like; "Standard" (all security-options active,no extra), "Light" (minim. libs etc. as little added as bare minimum?), "Security" (security tools,hardening,sec-related additions/code/guards/tools?), "Network" (security-add's same way but towards networking?) and maybe a hybrid of the last 2,"NetSec"(?)  adopted from open-source,free,reliable sources (or more diff pre-built sub-versions) where discussions internally and with main Zephyr-team to implement already established tools,policies,modifications for more security & flexibility and maybe rewrite some code to fit, audit and release - for example:

"ZephyrMember A" needs a an image for his Archer-router but need to deploy immideately in a very security-intense zone and might not have time configuring,compiling & lack time to build implement various tool's "ZephyrMember A" might want/need - to finally even have such a system like OpenWrt, where you can actually include what packages you need alongside for, say a "network-hardened" image with tools,libraries & pre-set policies already built for the members router-model and save time to deploy.

This way Zephyr-RTOS can reach a broader public whom may become contributors but also to be compatible with Zephyr founders guidelines & future aspect, where we can strive to be the bridge to easen up for non-advanced - semi-advanced users to get a chance also to see the Zephyr's possibilities plus we'll get alot more bug/testing/pen/battle-testing-reports & could increase overall usage by huge numbers where we can work towards a more secure & a little more user-friendly install in the end, but not limited to of course.

I'd like to hear your voices on this idea, both positive and negative of course , along with advices, tips, if this is even do'able and NOT conflicting with Zephyr-project's rules/guidelines or stance - rather needs to be in conjunction with dev-team's viewpoint on such a project & each new implement that the project-members of 'binaries' need to discuss with the main dev-team, to also receive input where a certain binary-build should/could improve on or lean towards,making it dynamic and flexible & not limited to, but be such as rules & guidelines may also need to be in check against dev-team's advice & wishes on what people want 3rd party library or tool should/could/wanted in a certain build to strive towards an easy install with detailed and approachable instruction - do you see the same possibilities with this as i do?

I see huge potential in such a project & dev's can concentrate on the Zephyr code-base whereas this potential project-party concentrate on delivering updated "binaries" (or maybe "installs" at a later stage) on each build with correct,secure build-flags,set's etc for a security-focus and alongside implement tools,hardening-techniques,policies,libraries whereas people NEEDING strong security but LACKING skill can be reached,also very talented programmers whom might've not heard about it or just avoided because no binary to try Zephyr's awesomeness - instead let's reach as many devices as possible with the differing binary-specfics!

Can we get a hand-raise to see how many would like to partake in such a project? [to get an estimate]
As i figure, we'd need roughly 35-40 dedicated people (for starters) with Zephyr-knowledge, need a couple with rigid,advanced coding-skills along of course with an understanding of the Zephyr-codebase, a few people need to take care of the packing and submissions, a few for analyzing bugs and compatibility-related problems reporting those backs to dev's,debugging & solving, implementation-reviewal team of finding new add's, a few website-maintainers & a few for a "think-tank"-people to brainstorm new ideas,feats in Zephyr-RTOS & overall and future of 'binary'-team's path, a few for releasing and discussing, a couple mods for a forum - especially bugs,compatibility-problems and issues or other to-be-dealt-with & automatically shared back to dev-team, making Zephyr development take a leap forward when such questions as forward/backward-compatibility (i.e) with i.e SIEM-system where very useful information the 'binary'-team can detect & creating a way for binary-crew to reach dev's to either give advice for/against & what to be looked-out for - that way the dynamic of such teamwork could have maybe a richer foresight in both project's way!

* Okaaay?
Can the Zephyr-founders & crew contemplate this idea & see the benefit of such an approach, where we could bring this to a whole new level?
The potential for broader reach and development in this hypothetical project is enourmous!

*
-[IF THIS IS IN ANYWAY AGAINST *ANY* RULES OR REGULATIONS,GUIDELINES, THIS POST CAN BE DELETED WITHOUT HESITATION]-
*

# Problem description

Getting a broader public to get to try Zephyr-RTOS with aim for a "Standard" install on - finally - all possible devices! 
What we want is a place to for starters give a "Standard"-flavour f(of Zephyr for all devices where the in-scope audience will find pre-built binaries for (hopefully) every device with easier installation (if possible) and a dynamic, updated and well-documented installation instruction & a usage instruction with features,changelog & what extra features each 'flavour' has added to it, for instance, 'Security'-image for a Acer Aspire model xxsome123.4 , 'Network'-image suitable for extra security in network or maybe a 'Hybrid'-flavour of both?

# Proposed change

A new start for great expansion!

# Detailed RFC

Needs to be set for;
1, Project's goal and to be acknowledged by head-team & dev-team of Zephyr.
2, Project should always strive for improved security and compatibility.
3, A shared platform where dev's can share thought's and request's on 'binary-team' and vice versa - a complete sharing symbiose bidirectional communication and advice/requests/what-not/what-ever in a for instance, closed forum?
4, Giving a broad public a easy way to get to use a pre-documented, pre-compiled image of security with compatibility,speed and great coverage with easy-understood documentation to their specific device & model!

# Proposed (Detailed)

Approach to severely impact the general market by increasing security-measures for an audience lacking the skills for even trying - to a humble, easy installation on each platform, where different devices have different needs! 
Ability to expand this project, to tighten security & compatibility further & to give a powerful Zephyr-RTOS easily installable for many more people by offering a library of pre-built binaries designed for specific model of specific device with a variance & maybe at last even a completely  completely customizable installation whereas user may interactively install modules, options and libraries on top of Zephyr by 3rd-party,open-source,reliable sources which has been approved and put together by the 'binary'-team but approved additions by head-team of Zephyr and binary-team members.

# Dependencies

Dep's needs to be addressed as they develop.

# Concerns and Unresolved Questions

Concern is, does such a project conflict with Zephyr-RTOS head or dev-team's rules,guidelines or anything other rendering such a project impossible?

# Alternatives

Doesn't...exist? Right?

---

*
-[IF THIS IS IN ANYWAY AGAINST *ANY* RULES OR REGULATIONS,GUIDELINES, THIS POST CAN BE DELETED WITHOUT HESITATION]-
*

All input is welcome!

**Comment by carlescufi** [comment]
@zt3ph4n a few points that you might find useful:

1. Providing binary builds of Zephyr does not go against the project's rules or guidelines, but it is just not practical (see next point)
2. Zephyr is not Linux. The application and the kernel are built and linked together, so providing precompiled builds makes little sense, because you cannot decouple the application from the kernel
3. You probably want to consider instead providing users with a hardened/secured version of the Zephyr codebase that they can then use to make products

**Reply by zt3ph4n** [reply]
Building model-specific images & work thwards a simplificated,universal method for installation and work out method's for a more general public, so to speak? Though i was specific enough.

**Comment by zt3ph4n** [comment]
> @zt3ph4n a few points that you might find useful:
> 
> 1. Providing binary builds of Zephyr does not go against the project's rules or guidelines, but it is just not practical (see next point)
> 2. Zephyr is not Linux. The application and the kernel are built and linked together, so providing precompiled builds makes little sense, because you cannot decouple the application from the kernel
> 3. You probably want to consider instead providing users with a hardened/secured version of the Zephyr codebase that they can then use to make products

Ill work out something useful to prove what i mean,i'll re-post with something more substantial,closing this thread, opening in main at a later point.
